# Example: housing price prediction using square footage and bedrooms and bathrooms, evaluation with std

In [1]:
import numpy as np 
import pandas as pd 
from sklearn.linear_model import LinearRegression
from sklearn.model_selection import train_test_split
from sklearn.metrics import mean_squared_error

import os
for dirname, _, filenames in os.walk('/kaggle/input'):
    for filename in filenames:
        print(os.path.join(dirname, filename))


/kaggle/input/house-prices/house-prices.csv


In [2]:
file_path = '/kaggle/input/house-prices/house-prices.csv'  
p = pd.read_csv(file_path)



In [3]:
target= p["Price"]
feature = p[["SqFt","Bedrooms","Bathrooms"]]
x_train,x_test,y_train,y_test=train_test_split(feature,target,test_size=0.33,shuffle=False)


In [4]:
lin_reg=LinearRegression()
lin_reg.fit(x_train,y_train)


LinearRegression()

In [5]:
y_predict=lin_reg.predict(x_test)
mse=mean_squared_error(y_predict,y_test)
std_test=np.sqrt(mse)

In [6]:
desc = target.describe()
std_value = desc['std']
print(f"Standard Deviation total: {std_value} & Standard Deviation Test:{std_test} ")

Standard Deviation total: 26868.770370734055 & Standard Deviation Test:21919.174288659113 


In [7]:
submission = pd.DataFrame({'SalePrice real': y_test, 'SalePrice predict': y_predict})
submission.to_csv('/kaggle/working/submission.csv')
print("The submission file has been created successfully!")

The submission file has been created successfully!


# Result: The closer the standard deviation total and test are, the better
Standard Deviation total: 26868.770370734055 & Standard Deviation Test:21919.174288659113